ako-gateway-api/nodes/avi_model_l7_translator.go

```diff
@@ -16,30 +16,39 @@ package nodes
 
 import (
 	"fmt"
+	"strconv"
+	"strings"
 
 	"github.com/vmware/alb-sdk/go/models"
 	"google.golang.org/protobuf/proto"
 
+	avimodels "github.com/vmware/alb-sdk/go/models"
+	akogatewayapilib "github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/ako-gateway-api/lib"
+	akogatewayapiobjects "github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/ako-gateway-api/objects"
 	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/lib"
 	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/nodes"
 	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/pkg/utils"
 )
 
-func (o *AviObjectGraph) ProcessL7Routes(key string, routeModel RouteModel, parentNsName string) {
+func (o *AviObjectGraph) ProcessL7Routes(key string, routeModel RouteModel, parentNsName string, listeners []string) {
 	for _, rule := range routeModel.ParseRouteRules().Rules {
 
 		// TODO: add the scenarios where we will not create child VS here.
```

Comment:
We may have to do some experiments for HTTPRoute with filter and without backend or match.